### PR TITLE
Fix Sphinx/RTD issues

### LINKED ISSUE #5230: Failing RTD build: "No module named matrix_utilities"
There is build failure on RTD, though I'm not sure why it started to show up just now. Maybe a wipe and rebuild will be enough.

https://readthedocs.org/projects/astropy/builds/4284616/

```
Traceback (most recent call last):
  File "/home/docs/checkouts/readthedocs.org/user_builds/astropy/conda/latest/lib/python2.7/site-packages/Sphinx-1.3.5-py2.7.egg/sphinx/cmdline.py", line 244, in main
    app.build(opts.force_all, filenames)
  File "/home/docs/checkouts/readthedocs.org/user_builds/astropy/conda/latest/lib/python2.7/site-packages/Sphinx-1.3.5-py2.7.egg/sphinx/application.py", line 284, in build
    self.emit('build-finished', None)
  File "/home/docs/checkouts/readthedocs.org/user_builds/astropy/conda/latest/lib/python2.7/site-packages/Sphinx-1.3.5-py2.7.egg/sphinx/application.py", line 508, in emit
    results.append(callback(self, *args))
  File "/home/docs/checkouts/readthedocs.org/user_builds/astropy/conda/latest/lib/python2.7/site-packages/sphinx_gallery/gen_gallery.py", line 209, in sumarize_failing_examples
    "\n" + "-" * 79)
ValueError: Here is a summary of the problems encountered when running the examples

Unexpected failing examples:
../examples/coordinates/plot_sgr-coordinate-frame.py failed leaving traceback:
Traceback (most recent call last):
  File "/home/docs/checkouts/readthedocs.org/user_builds/astropy/conda/latest/lib/python2.7/site-packages/sphinx_gallery/gen_rst.py", line 520, in execute_code_block
    exec(code_block, example_globals)
  File "<string>", line 3, in <module>
ImportError: No module named matrix_utilities


-------------------------------------------------------------------------------

Exception occurred:
  File "/home/docs/checkouts/readthedocs.org/user_builds/astropy/conda/latest/lib/python2.7/site-packages/sphinx_gallery/gen_gallery.py", line 209, in sumarize_failing_examples
    "\n" + "-" * 79)
ValueError: Here is a summary of the problems encountered when running the examples

Unexpected failing examples:
../examples/coordinates/plot_sgr-coordinate-frame.py failed leaving traceback:
Traceback (most recent call last):
  File "/home/docs/checkouts/readthedocs.org/user_builds/astropy/conda/latest/lib/python2.7/site-packages/sphinx_gallery/gen_rst.py", line 520, in execute_code_block
    exec(code_block, example_globals)
  File "<string>", line 3, in <module>
ImportError: No module named matrix_utilities
``

### PR DESCRIPTION
**Pull request by astrofrog**
This is an attempt to fix #5230 and #5244 

Am testing RTD on https://readthedocs.org/projects/astrofrog-debug/builds/

### CODE REVIEW

**Comment by astrofrog** [comment]
Ok so changing to 0.1.2 on RTD doesn't fix the issue....

**Comment by astrofrog** [comment]
Ok, I've managed to reproduce the RTD issue on Travis! Baby steps!

**Comment by astrofrog** [comment]
I'm wondering if this is somehow because we install astropy and then build with sphinx-build rather than use ``python setup.py build_docs``

**Comment by astrofrog** [comment]
I think I know!

**Comment by pllim** [comment]
Wow, good investigative work!

**Comment by astrofrog** [comment]
This passes on RTD. If it also passes on Travis then I'll merge in a couple of days. The only slightly hacky thing is temporarily installing wcsaxes during the docs build if not already present.

**Comment by astrofrog** [comment]
This does fix both #5230 and #5244 by the way.

**Comment by astrofrog** [comment]
I'm milestoning this to v1.0.11 since I think it may have to be backported all the way.

File: .rtd-environment.yml [hunk 1]
```diff
@@ -1,17 +1,21 @@
...
-  - wcsaxes
   - scipy
   - pillow
   - pip:
-    - sphinx-gallery
-    - jplephem
\ No newline at end of file
+    - sphinx-gallery>=0.1.3
+    - jplephem
+    - astropy-helpers
```
Comment (mhvk) [inline .rtd-environment.yml]: missing end of line

File: docs/conf.py [hunk 2]
```diff
@@ -39,6 +42,16 @@
     # If that doesn't work trying to import from astropy_helpers below will
     # still blow up
 
+# We now check for any dependencies that are required to build the docs that
+# depend on Astropy, since these may not be installed yet. For instance, on
+# ReadTheDocs, we can't set up wcsaxes with conda since that would result in the
+# astropy conda package getting installed, which would shadow the developer
+# version installed just prior to building the docs. So we should set up any
+# such dependencies here.
+if ON_RTD:
```
Comment (mhvk) [inline docs/conf.py]: Maybe define a `ON_TRAVIS` here too and use that below?

**Comment by mhvk** [comment]
Nice detective work indeed! My comments are so minor you should feel free to ignore them.

File: .rtd-environment.yml (hunk 1 above)
Comment (astrofrog) [inline .rtd-environment.yml]: Fixed

File: docs/conf.py (hunk 2 above)
Comment (astrofrog) [inline docs/conf.py]: Fixed

File: .rtd-environment.yml (hunk 1 above)
Comment (bsipocz) [inline .rtd-environment.yml]: does ``astropy-helpers`` needed to be listed?

Comment (astrofrog) [inline .rtd-environment.yml]: Strictly speaking I think not. I'll try without.

File: docs/conf.py [hunk 3]
```diff
@@ -229,7 +243,8 @@
             'astropy': None,
             'matplotlib': 'http://matplotlib.org/',
             'numpy': 'http://docs.scipy.org/doc/numpy/',
-        }
+        },
+        'abort_on_example_error': False
```
Comment (Cadair) [inline docs/conf.py]: This probably only wants to be false on RTD? You want a travis build to fail if the examples are failing??

**Comment by astrofrog** [comment]
@Cadair - did you comment on this? I got a notification but don't see any comment :)

**Comment by Cadair** [comment]
@astrofrog yeah, I missed something though, so it's irrelevant.

Out of interest though, would you rather the RTD build fail or have broken gallery examples in the final build? You went for the latter and I kinda see why, I am interested in your logic though.

**Comment by astrofrog** [comment]
I guess I see your point that we should probably fail RTD too...

File: docs/conf.py [hunk 4]
```diff
@@ -256,8 +256,6 @@
     # sphinx-gallery: https://github.com/sphinx-gallery/sphinx-gallery/pull/97
```
Comment (Cadair) [inline docs/conf.py]: this comment can be removed?

Comment (astrofrog) [inline docs/conf.py]: Fixed

**Comment by mhvk** [comment]
Looks all OK, so merging!

**Comment by mhvk** [comment]
Oops, added label...

**Comment by eteq** [comment]
pretty sure this is actually v1.2.2 instead of 1.0.11, because all the changes here seem to not be relevant to 1.0.11 ... but if 1.0.x needs some of this we can apply it manually if the builds are failing for some reason.

**Pull request merged**